src/SFML/System/Alloc.cpp

```diff
@@ -0,0 +1,44 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2023 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Alloc.h>
+
+#include <stdlib.h>
+
+
+////////////////////////////////////////////////////////////
+void* sfMalloc(size_t size)
+{
+    return malloc(size);
+}
+
+
+////////////////////////////////////////////////////////////
+void sfFree(void* ptr)
+{
+    free(ptr);
+}
```

Comment:
Why do we need to provide functions that wrap existing C standard library functions? Why can't users just call `free` themselves?

Reply:
C# for example doesn't have access to the C standard library functions (as far as I'm aware). There is `Marshal.AllocHGlobal` and `Marshal.FreeHGlobal` but I don't think they are the same as `malloc` and `free`. Therefore, unless CSFML exports a method to free the pointers that can be P/Invoked from C#, there's no way for the C# bindings to use methods like `sfKeyboard_getDescription` without memory leaks.

Reply:
That's fair. If our API provides an allocation, we must provide a way to deallocate and if users can't use the C standard library then I suppose we're forced to add `sfFree`. Let's remove `sfMalloc` though since users should have no reasons to use that and we don't need it internally. We add it later should the need arise.

Reply:
Sound good, I've pushed an update with `sfMalloc` removed.